### PR TITLE
feat: Add adapter interface based on connector-api documentation

### PR DESCRIPTION
**Pull request by ghost**
Includes example project to demonstrate how the adapter works

### CODE REVIEW

File: src/backend/GnuCashBudget.Adapter.Abstractions/Models/Expense.cs [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+public class Expense
+{
+    public decimal Amount { get; set; }
+    public string Currency { get; set; }
+    public string? Category { get; set; }
+    public string? Description { get; set; }
+    public string Timestamp { get; set; }
+    public string? Merchant { get; set; }
+    public string? PaymentMethod { get; set; }
```
Comment (veselfil) [inline src/backend/GnuCashBudget.Adapter.Abstractions/Models/Expense.cs]: enum?

File: src/backend/GnuCashBudget.Adapter.Example/Services/AdapterService.cs [hunk 2]
```diff
@@ -0,0 +1,18 @@
+using GnuCashBudget.Adapter.Abstractions.Models;
+using GnuCashBudget.Adapter.Example.Configuration;
+using Microsoft.Extensions.Options;
+
+namespace GnuCashBudget.Adapter.Example.Services;
+
+public class AdapterService(IOptions<ExampleOptions> exampleOptions) : IAdapterService
```
Comment (veselfil) [inline src/backend/GnuCashBudget.Adapter.Example/Services/AdapterService.cs]: We probably don't need this class if we just registered IAdapter to DI and used it directly in the controller

**Review by veselfil — CHANGES_REQUESTED** [review]

File: src/backend/GnuCashBudget.Adapter.Abstractions/Models/Expense.cs (hunk 1 above)
Comment (ghost) [reply src/backend/GnuCashBudget.Adapter.Abstractions/Models/Expense.cs]: Changed in 736d262a624db46800a33e95484b24f4955ef831

File: src/backend/GnuCashBudget.Adapter.Example/Services/AdapterService.cs (hunk 2 above)
Comment (ghost) [reply src/backend/GnuCashBudget.Adapter.Example/Services/AdapterService.cs]: IAdapterService was there only as a showcase how Adaptee and Adapter are working together. After discussion I removed it and used IBankTarget directly in controller. Changes are here: 05689137a0507e5fbcafebe62ef6205c9e55483a

File: src/backend/GnuCashBudget.Adapter.Abstractions/Models/AdapterResponse.cs [hunk 3]
```diff
@@ -0,0 +1,7 @@
+namespace GnuCashBudget.Adapter.Abstractions.Models;
+
+public class AdapterResponse
+{
+    public string? ContinuationToken { get; set; }
+    public IEnumerable<Expense> Expenses { get; set; }
```
Comment (veselfil) [inline src/backend/GnuCashBudget.Adapter.Abstractions/Models/AdapterResponse.cs]: Let's generalize this to `Transactions` or something

**Review by veselfil — CHANGES_REQUESTED** [review]

File: src/backend/GnuCashBudget.Adapter.Abstractions/Models/AdapterResponse.cs (hunk 3 above)
Comment (ghost) [reply src/backend/GnuCashBudget.Adapter.Abstractions/Models/AdapterResponse.cs]: done

**Review by veselfil — APPROVED** [review]

**Pull request merged**